ideas/293-ulc-integration/README.md

```diff
@@ -0,0 +1,101 @@
+---
+id: 293-ulc-integration
+title: ULC integration
+status: draft
+created: 2018-08-06
+category: core
+lead-contributor: b00ris
+contributors:
+    - b00ris Go dev
+    - jeka Go dev
+exit-criteria: no
+success-metrics: no
+clear-roles: no
+future-iteration: no
+roles-needed:
+    - Go dev
+    - Go dev
+    - Clojure dev
+    - QA
+okrs:
+    - Core. [P0] LES and/or ULC are operational and used by at least 10% of all users.
+    - Research. Integrate and test ULC in a manner “graceful downgrade”: use infura by default and if it fails, start ULC node.
+---
+
+## Preamble
+
+    Idea: 293-ulc-integration
+    Title: ULC integration
+    Status: Draft
+    Created: 2018-08-06
+    Requires (*optional): #254
+
+## Summary
+Status app can use LES and ULC as Ethereum providers instead of Infura.
+
+## Swarm Participants
+- Lead Contributor: @b00ris
+- Contributor: @jeka
+- Contributor: @mandrigin
+- Contributor: TBD
+- QA: TBD
+
+## Product Overview
+We need LES client enabled on mobile device to be really decentralized and to enable all web3 features for dApps.
+
+### Requirements & Dependencies
+Depends on research swarm #254
+
+### Security and Privacy Implications
+* Free slots for LES servers
+* Trusted nodes selection
+
+### Minimum Viable Product
+* LES can work as the second Ethereum provider in status app.
+* ULC can work as the third Ethereum provider in status app with static trusted nodes.
```

Comment:
I think MVP is just LES, but perhaps the other iteration re ULC is concurrent? So we are working on both at the same time.

I don't understand why the goal date is in October? That's way too far away for initial LES (even if bad perf/a bit buggy) cc @mandrigin

Reply:
If we are taking "just LES" as a first iteration (and in WIP), it should be done by Sep, 1st, I think.
Later, we can estimate how broken is it and what the next iterations are. Right now I know too little about the state of LES and ULC to make estimates for anything besides that.